pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -331,6 +331,12 @@ private TableAuthorizationResult hasTableAccess(RequesterIdentity requesterIdent
    * Returns true if the QPS quota of the tables has exceeded.
    */
   private boolean hasExceededQPSQuota(Set<String> tableNames, RequestContext requestContext) {
+    String database = DatabaseUtils.extractDatabaseFromTableName(tableNames.iterator().next());
```

Comment:
The `database` has already been extracted from the request when this method is called - probably better to pass that in rather than re-extracting it from the table name?

Reply:
Ah, and also this list of tables can be empty in literal only queries - see https://github.com/apache/pinot/blob/1d788196d31793fe7209506c52b7c4424a43c0b3/pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java#L634 which is failing in CI due to this.

Reply:
Ah right, let me handle this. Nice catch!